src/github_client.py

```diff
@@ -80,6 +81,38 @@ def get_runner_applications(self, path: GithubPath) -> RunnerApplicationList:
 
         return runner_bins
 
+    @catch_http_errors
+    def get_runner_application(
+        self, path: GithubPath, arch: Arch, os: str = "linux"
+    ) -> RunnerApplication:
+        """Get runner application available for download for given arch.
+
+        Args:
+            path: GitHub repository path in the format '<owner>/<repo>', or the GitHub organization
+                name.
+            arch: The runner architecture.
+            os: The operating system that the runner binary should run on.
+
+        Raises:
+            RunnerBinaryError: If the runner application for given architecture and OS is not
+                found.
+
+        Returns:
+            The runner application.
+        """
+        runner_applications = self.get_runner_applications(path=path)
```

Comment:
Could this function be private/ removed and moved into this function now?

Reply:
Refactored, thank you!